apps/crn-server/test/routes/research-outputs.route.test.ts

```diff
@@ -272,37 +272,92 @@ describe('/research-outputs/ route', () => {
       );
     });
 
-    test('Should return a validation error when the authors data is not valid', async () => {
+    describe('Authors validation', () => {
       const researchOutput = getCreateResearchOutput();
-      researchOutputControllerMock.create.mockRejectedValueOnce(
-        Boom.badRequest(),
-      );
+      const fields = ['userId', 'externalAuthorId', 'externalAuthorName'];
 
-      const response = await supertest(app)
-        .post('/research-outputs')
-        .send({
-          ...researchOutput,
-          authors: [{ userId: '222', externalAuthorId: '333' }],
-        })
-        .set('Accept', 'application/json');
+      test('Should return a validation error when required field is missing', async () => {
+        researchOutputControllerMock.create.mockRejectedValueOnce(
+          Boom.badRequest(),
+        );
 
-      expect(response.status).toBe(400);
-      expect(response.body).toEqual({
-        error: 'Bad Request',
-        message: 'Validation error',
-        statusCode: 400,
-        data: [
-          {
-            instancePath: '/authors/0',
-            schemaPath: '#/properties/authors/items/oneOf',
-            keyword: 'oneOf',
-            params: {
-              passingSchemas: [0, 1],
+        const response = await supertest(app)
+          .post('/research-outputs')
+          .send({
+            ...researchOutput,
+            authors: [{ name: 'random-value' }],
+          })
+          .set('Accept', 'application/json');
+
+        expect(response.status).toBe(400);
+        expect(response.body).toEqual({
+          error: 'Bad Request',
+          message: 'Validation error',
+          statusCode: 400,
+          data: [
+            ...fields.map((field, idx) => ({
+              instancePath: '/authors/0',
+              schemaPath: `#/properties/authors/items/oneOf/${idx}/required`,
+              keyword: 'required',
+              params: {
+                missingProperty: field,
+              },
+              message: `must have required property '${field}'`,
+            })),
+            {
+              instancePath: '/authors/0',
+              keyword: 'oneOf',
+              message: 'must match exactly one schema in oneOf',
+              params: {
+                passingSchemas: null,
+              },
+              schemaPath: '#/properties/authors/items/oneOf',
             },
-            message: 'must match exactly one schema in oneOf',
-          },
-        ],
+          ],
+        });
       });
+
+      test.each(fields)(
+        'Should return a validation error when passing more than one required fields (%s)',
```

Comment:
i dont understand this test, you seem to override one of each of the property values in the author object with 'id-1'.

What I asked previously was to make sure that you can only send:
```typescript
{
   userId: 'user-id'
}
```
OR
```typescript
{
   externalAuthorId: 'external-author-uid'
}
```
OR
```typescript
{
   externalAuthorName: 'external-author-name'
}
```
So you should not be able to send
```typescript
{
   somethingRandom: 'external-author-name'
}
```
which you tested correctly
OR
```typescript
{
   externalAuthorName: 'external-author-name',
   externalAuthorId: 'author-1',
}
```
which you have not, similarily you should not be able to send:
```typescript
{
   userId: 'external-author-name',
   externalAuthorId: 'author-1',
}
```
OR
```typescript
{
   userId: 'external-author-name',
   externalAuthorName: 'author-1',
}
```
OR
```typescript
{
   userId: 'external-author-name',
externalAuthorId: 'author-1',
   externalAuthorName: 'author-1',
}
```

does it make sense?